### PR TITLE
Rework model for circuit inspections. Add inspect support back to mgm…

### PR DESCRIPTION
**Pull request by plorenz**
…t channel

### CODE REVIEW

File: controller/api_impl/inspections_api_model.go [hunk 1]
```diff
@@ -29,10 +31,27 @@ func MapInspectResultToRestModel(inspectResult *network.InspectResult) *rest_mod
 		Success: &inspectResult.Success,
 	}
 	for _, val := range inspectResult.Results {
+		var emitVal interface{}
+		if strings.HasPrefix(val.Value, "{") {
+			mapVal := map[string]interface{}{}
```
Comment (ekoby) [inline controller/api_impl/inspections_api_model.go:36]: is `HasPrefix()` preferable to `if val.Value[0] == '{'`

File: router/xgress/link_receive_buffer.go [hunk 2]
```diff
@@ -84,9 +85,9 @@ func (buffer *LinkReceiveBuffer) getLastBufferSizeSent() uint32 {
 	return atomic.LoadUint32(&buffer.lastBufferSizeSent)
 }
 
-func (buffer *LinkReceiveBuffer) Inspect() map[string]interface{} {
-	result := map[string]interface{}{}
-	result["size"] = buffer.Size()
-	result["lastSizeSent"] = buffer.getLastBufferSizeSent()
-	return result
+func (buffer *LinkReceiveBuffer) Inspect() inspect.XgressRecvBufferDetail {
```
Comment (ekoby) [inline router/xgress/link_receive_buffer.go:88]: return `*inspect.XgressRecvBufferDetail` for consistency with `LinkSendBuffer.Inspect()`

File: router/xgress/xgress.go [hunk 3]
```diff
@@ -580,16 +581,15 @@ func (self *Xgress) SendEmptyAck() {
...
+func (self *Xgress) InspectCircuit(detail *inspect.CircuitInspectDetail) {
 	timeSinceLastRxFromLink := time.Duration(info.NowInMilliseconds()-atomic.LoadInt64(&self.timeOfLastRxFromLink)) * time.Millisecond
-	result["timeSinceLastLinkRx"] = timeSinceLastRxFromLink.String()
-	result["sendBuffer"] = self.payloadBuffer.Inspect()
-	result["recvBuffer"] = self.linkRxBuffer.Inspect()
+	xgressDetail := &inspect.XgressDetail{
+		Address:             string(self.address),
+		Originator:          self.originator.String(),
+		TimeSinceLastLinkRx: timeSinceLastRxFromLink.String(),
+		SendBufferDetail:    *self.payloadBuffer.Inspect(),
```
Comment (ekoby) [inline router/xgress/xgress.go:590]: any reason to not have detail fields pointers?

**Review by ekoby — APPROVED** [review]

File: router/xgress/link_receive_buffer.go (hunk 2 above)
Comment (plorenz) [reply router/xgress/link_receive_buffer.go]: Fixed

File: router/xgress/xgress.go (hunk 3 above)
Comment (plorenz) [reply router/xgress/xgress.go]: I liked guaranteeing they wouldn't be null, but they're always going to be non-nil just based on usage, so probably makes sense to just use pointers. Updated.

**Pull request merged**